src/components/GroupForm.js

```diff
@@ -29,31 +31,43 @@ export default connect(mapStateToProps)(function GroupForm ({users, userId, memb
           </Item>
         </Form>
 
-        <Form>
-            <Item>
-              <Input placeholder='Search Users'/>
-            </Item>
-        </Form>
+        <Item>
+          <List dataArray={tempList}
+            renderRow={user =>
+              <ListItem>
+                <Text>{user}</Text>
+              </ListItem>
+            }>
+          </List>
+        </Item>
 
-        <List dataArray={users}
-          renderRow={user =>
-            <ListItem key={user.id} 
-            onPress={() => dispatch({type: 'ADD_MEMBER', id: user.id}) }>
-              <Text>{`${user.first_name} ${user.last_name}`}</Text>
-            </ListItem>
-          }>
-        </List>
+        <Item>
+          <Input onChangeText={text => dispatch({type: 'SEARCH_NAME', text: text})} placeholder='Add a Member'/>
+          <List dataArray={searchResults}
+            renderRow={user =>
+              <ListItem onPress={() => {
+                dispatch({type: 'ADD_MEMBER', id: user.id});
+              }}>
+                <Text>{user.username}</Text>
+              </ListItem>
+            }>
+          </List>
+        </Item>
 
       </Content>
 
       <Button
         style={{ alignSelf: 'center', marginTop: 20, marginBottom: 20 }}
-        onPress={()=> {
+        onPress={() => {
           let data = {
             name: groupName,
             members: [...members, userId ]
           };
-          postGroup(data).then(data => getGroups(dispatch).then(() => Actions.groups()));
+          postGroup(data)
```

Comment:
The only thing changed here what that I put the then statements on new lines, it's not showing that in here for some reason, but has changed in the actual code.